charts/capi-runtime-extensions/defaultclusterclasses/nutanix-cluster-class.yaml

```diff
@@ -0,0 +1,708 @@
+                        value: "true"
+                      - name: vip_leaseduration
+                        value: "15"
+                      - name: vip_renewdeadline
+                        value: "10"
+                      - name: vip_retryperiod
+                        value: "2"
+                      - name: svc_enable
+                        value: "${KUBEVIP_SVC_ENABLE=false}"
+                      - name: lb_enable
+                        value: "${KUBEVIP_LB_ENABLE=false}"
+                      - name: enableServicesElection
+                        value: "${KUBEVIP_SVC_ELECTION=false}"
+                    securityContext:
+                      capabilities:
+                        add:
+                          - NET_ADMIN
+                          - SYS_TIME
+                          - NET_RAW
+                    volumeMounts:
+                      - mountPath: /etc/kubernetes/admin.conf
+                        name: kubeconfig
+                    resources: {}
+                hostNetwork: true
+                hostAliases:
+                  - hostnames:
+                      - kubernetes
+                    ip: 127.0.0.1
+                volumes:
+                  - name: kubeconfig
+                    hostPath:
+                      type: FileOrCreate
+                      path: /etc/kubernetes/admin.conf
+              status: {}
+            owner: root:root
+            path: /etc/kubernetes/manifests/kube-vip.yaml
+        initConfiguration:
+          nodeRegistration:
+            kubeletExtraArgs:
+              cloud-provider: external
+              eviction-hard: nodefs.available<10%,nodefs.inodesFree<5%,imagefs.available<15%,memory.available<100Mi,imagefs.inodesFree<10%
+              tls-cipher-suites: ${TLS_CIPHER_SUITES=TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256}
+        joinConfiguration:
+          nodeRegistration:
+            kubeletExtraArgs:
+              cloud-provider: external
+              eviction-hard: nodefs.available<10%,nodefs.inodesFree<5%,imagefs.available<15%,memory.available<100Mi,imagefs.inodesFree<10%
+              tls-cipher-suites: ${TLS_CIPHER_SUITES=TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256}
+        postKubeadmCommands:
+          - echo export KUBECONFIG=/etc/kubernetes/admin.conf >> /root/.bashrc
+          - echo "after kubeadm call" > /var/log/postkubeadm.log
+        preKubeadmCommands:
+          - echo "before kubeadm call" > /var/log/prekubeadm.log
+          - hostnamectl set-hostname "{{ ds.meta_data.hostname }}"
```

Comment:
This might have to be modified to accommodate 1.29 https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/pull/382

Reply:
currently we are using alpha1 version. if we cut alpha2, then latest changes will show up

Reply:
handled in https://github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/pull/6